File: app/src/main/java/br/com/ufg/www/events/mvp/login/Contract.java
package br.com.ufg.www.events.mvp.login;

import br.com.ufg.www.events.domain.UIFeedback;
import br.com.ufg.www.events.model.Login;

public interface Contract {

    interface View extends UIFeedback {
        void login();

        void onLoggedIn();
    }

    interface Presenter {
        void login(Login login);
    }

    interface Interactor {
        void login(Login login);
    }

}

File: app/src/main/java/br/com/ufg/www/events/mvp/user/login/Contract.kt
package br.com.ufg.www.events.mvp.user.login

import br.com.ufg.www.events.domain.UIFeedback
import br.com.ufg.www.events.model.Login

interface Contract {

    interface View : UIFeedback {
        fun login()
        fun register()
        fun onLoggedIn()
        fun loginFailed()
    }

    interface Presenter {
        fun login(login: Login)
    }

}